src/main/java/com/guciowons/shoppingbasket/Product/ProductService.java

```diff
@@ -11,20 +11,50 @@
 public class ProductService {
 
     private final ProductClient productClient;
+    private final ProductRepository productRepository;
 
-    public ProductService(ProductClient productClient) {
+    public ProductService(ProductClient productClient, ProductRepository productRepository) {
         this.productClient = productClient;
+        this.productRepository = productRepository;
     }
 
-    public List<Product> getProducts() throws NoExternalConnectionException {
+    @Scheduled(fixedDelay = 3600000)
+    public void refreshProductsInDatabase(){
         try {
-            return productClient.getProducts();
-        }catch(FeignException e){
-            throw new NoExternalConnectionException("Cant connect with external api");
+            insertProducts();
+        }catch (FeignException e){
+            System.out.println("Error connecting to external api. Products will be downloaded in one hour");
         }
     }
 
-    public Optional<Product> getProductById(int productId){
-        return productClient.getProductById(productId);
+    public void insertProducts(){
+        if (productRepository.findAll().isEmpty()) {
```

Comment:
To sprawdzenie nie ma sensu, bo i tak jeśli nie znajdziesz istniejącego produktu z ID to robisz zapis nowego. Wywołanie `updateProducts()` wystarczy.